service/src/main/java/bio/terra/workspace/service/iam/model/WsmIamRole.java

```diff
@@ -46,20 +50,23 @@ public static WsmIamRole fromSam(String samRole) {
         .orElse(null);
   }
 
-  public static WsmIamRole getHighestRole(UUID workspaceId, List<WsmIamRole> roles) {
+  public static Optional<WsmIamRole> getHighestRole(UUID workspaceId, List<WsmIamRole> roles) {
     if (roles.isEmpty()) {
-      throw new InternalServerErrorException(
-          String.format("Workspace %s missing roles", workspaceId.toString()));
+      // This should be extremely rare. This only happens when a WSM role has been added to SAM,
+      // but WSM doesn't know about it yet (eg a local WSM created a workspace with this role, but
+      // broad-dev WSM doesn't know about role yet).
```

Comment:
I think the general case is that it happens when there is a workspace that has no WSM roles. Technically, it is not specific to us adding a Sam role for WSM. Suggest the warning below say "Ignoring workspace %s with no WSM roles".

Reply:
SAM listResourcesAndPoliciesV2 only returns workspaces where caller is in at least one role.

The exact sequence of events:

- With Local WSM, I created a workspace with discoverer policy.
- Local WSM grantRole, added Liam test user to discoverer policy
- Nightly test calls ListWorkspaces as Liam. Workspace is returned with only discoverer role.